File: MapServer/Data/Models/Location.cs
using System.ComponentModel.DataAnnotations;

namespace MapServer.Data.Models;

public class Location
{
    public Guid Id { get; init; }
    [MaxLength(255)] public string Street { get; init; } = string.Empty;
    [MaxLength(255)] public string StreetNumber { get; init; } = string.Empty;
    [MaxLength(255)] public string City { get; init; } = string.Empty;
    public double Longitude { get; init; }
    public double Latitude { get; init; }
    [MaxLength(255)] public string? Suburb { get; init; }
    [MaxLength(255)] public string? PostalCode { get; init; }

    // Navigation properties
    public ICollection<Company> Companies { get; init; } = [];
}

File: map-api-node/src/data/models/Location.ts
import { Entity, PrimaryGeneratedColumn, Column } from "typeorm";

@Entity()
export class Location {
  @PrimaryGeneratedColumn()
  id: number;

  @Column({ length: 255 })
  street: string;

  @Column({ length: 255 })
  streetNumber: string;

  @Column({ length: 1023 })
  city: string;

  @Column("decimal", { precision: 12, scale: 9 })
  longitude: number;

  @Column("decimal", { precision: 12, scale: 9 })
  latitude: number;

  @Column({ nullable: true, length: 1023 })
  suburb?: string;

  @Column({ nullable: true, length: 1023 })
  postalCode: string;
}
